File: voicesmith/src/main/cpp/de/jurihock/voicesmith/etc/Log.cpp
#include <voicesmith/etc/Log.h>

#include <android/log.h>

INITIALIZE_EASYLOGGINGPP

class AndroidLogHandler final : public el::LogDispatchCallback {

public:

  void handle(const el::LogDispatchData* data) {
    static const auto tag = "voicesmith.cpp";
    const auto msg = data->logMessage()->message().c_str();

    const el::Level src = data->logMessage()->level();
    std::optional<int> dst = std::nullopt;

    switch (src) {
      case el::Level::Verbose:
        dst = ANDROID_LOG_VERBOSE;
        break;
      case el::Level::Debug:
        dst = ANDROID_LOG_DEBUG;
        break;
      case el::Level::Info:
        dst = ANDROID_LOG_INFO;
        break;
      case el::Level::Warning:
        dst = ANDROID_LOG_WARN;
        break;
      case el::Level::Error:
        dst = ANDROID_LOG_ERROR;
        break;
      case el::Level::Fatal:
        dst = ANDROID_LOG_FATAL;
        break;
      default:
        break;
    }

    if (dst) {
      __android_log_print(dst.value(), tag, "%s", msg);
    }
  }

  static bool install() {
    el::Helpers::uninstallLogDispatchCallback<el::base::DefaultLogDispatchCallback>("DefaultLogDispatchCallback");
    return el::Helpers::installLogDispatchCallback<AndroidLogHandler>("AndroidLogHandler");
  }

};

static bool IsAndroidLogHandlerInstalled = AndroidLogHandler::install();

File: voicesmith/src/main/java/de/jurihock/voicesmith/etc/Log.kt
package de.jurihock.voicesmith.etc

typealias AndroidLog = android.util.Log

class Log {
  companion object {

    const val TAG = "voicesmith.java"
    const val DEBUG = AndroidLog.DEBUG
    const val INFO = AndroidLog.INFO
    const val WARN = AndroidLog.WARN
    const val ERROR = AndroidLog.ERROR

    fun d(msg: String) {
      AndroidLog.d(TAG, msg)
    }

    fun i(msg: String) {
      AndroidLog.i(TAG, msg)
    }

    fun w(msg: String) {
      AndroidLog.w(TAG, msg)
    }

    fun e(msg: String) {
      AndroidLog.e(TAG, msg)
    }

    fun e(msg: String, thr: Throwable) {
      AndroidLog.e(TAG, msg, thr)
    }

    fun e(thr: Throwable) {
      AndroidLog.e(TAG, thr.message, thr)
    }

    fun p(prio: Int, msg: String) {
      when(prio) {
        DEBUG -> AndroidLog.d(TAG, msg)
        INFO -> AndroidLog.i(TAG, msg)
        WARN -> AndroidLog.w(TAG, msg)
        ERROR -> AndroidLog.e(TAG, msg)
        else -> throw IllegalArgumentException(
          "Invalid log priority ${prio}!")
      }
    }

  }
}
